binlog_streamer.go

```diff
@@ -230,6 +231,22 @@ func (s *BinlogStreamer) Run() {
 			// the full query that was executed on the master (with annotations)
 			// that is otherwise not possible to reconstruct
 			query = ev.Event.(*replication.RowsQueryEvent).Query
+		case *replication.QueryEvent:
+			query = ev.Event.(*replication.QueryEvent).Query
+			if string(query) == "BEGIN" {
+				break
+			}
+			statement, err := sqlparser.Parse(string(query))
```

Comment:
Nitpick: I see `string(query)` twice here. Can we modify the above line to simply go: `query := string(e.Query)`?